clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1020,9 +1011,6 @@ private Future<RecordMetadata> doSend(ProducerRecord<K, V> record, Callback call
 
             if (result.abortForNewBatch) {
                 int prevPartition = partition;
-                // IMPORTANT NOTE: the following onNewBatch and partition calls should not interrupted to allow
-                // the custom partitioner to correctly track its state
-                onNewBatch(record.topic(), cluster, prevPartition);
```

Comment:
we don't need the path of "abortOnNewBatch", right? If so, could you please do a bit refactor for it?

Reply:
Thanks for the suggestion. Updated it.